libs/datamodel/core/tests/base/duplicates.rs

```diff
@@ -21,6 +21,37 @@ fn fail_on_duplicate_models() {
         Span::new(53, 57),
     ));
 }
+
+// From issue: https://github.com/prisma/prisma/issues/1988
+#[test]
+fn fail_on_duplicate_models_with_relations() {
+    let dml = r#"
+    model Post {
+      id Int @id
+    }
+
+    model Post {
+      id Int @id
+      categories Categories[]
+    }
+
+    model Categories {
+      post Post @relation(fields:[postId], references: [id])
+      postId Int
+    }
+    "#;
+
+    let errors = parse_error(dml);
+
+    errors.assert_length(2);
+    errors.assert_is_at(
+        0,
+        DatamodelError::new_duplicate_top_error("Post", "model", "model", Span::new(52, 56)),
+    );
+
+    errors.assert_is_at(1, DatamodelError::new_type_not_found_error("Post", Span::new(147, 151)));
```

Comment:
It is a bit weird that this additional error is shown. I think that this would not happen if you return early in the validation pipeline if any of the prechecks fail.